### DISCUSSION TITLE
How to set System Environment Variable as `NEXT_PUBLIC_` env?

### DISCUSSION THREAD

**Discussion by zoshigayan** [vercel/vercel#4422] [Help] [answered]
Hi.
I try to get current page's absolute URL in React Component with Next.js. It exported as a static site, so doesn't have any server-side logic.

I want to use now's [System Environment Variables](https://vercel.com/docs/v2/build-step#system-environment-variables) to get preview deployments' domain. However, any variables' names must have `NEXT_PUBLIC_` prefix to use [inlined environment variables](https://vercel.com/docs/v2/build-step#system-environment-variables) feature of Next.js.

How can I rename now's System Environment Variables?
I want to config as below:

```
NEXT_PUBLIC_URL=$VERCEL_URL
```

If is there any other solution to make it, please let me know...Thank you!

**Comment by styfle** [comment, marked as answer]
Hi @zoshigayan 

You'll need to add a mapping in `next.config.js` if you want to rename system environment variables.

https://nextjs.org/docs/api-reference/next.config.js/environment-variables

**Reply by zoshigayan** [reply]
Thank you @styfle .

> add a mapping in `next.config.js`

I didn't know the usecase writing functions in `next.config.js` , and I'm sure it work well!

https://github.com/zeit/next.js/blob/canary/examples/with-env-from-next-config-js/next.config.js

**Reply by styfle** [reply]
Just following up here, you no longer need to add system environment variables since they are automatically available with the `NEXT_PUBLIC_` prefix.

See the announcement: https://vercel.com/changelog/environments-variables-per-git-branch

See the docs: https://vercel.com/docs/environment-variables#:~:text=prefixed%20Environment%20Variables
